code/Data_Analysis.ipynb

```diff
@@ -1096,143 +1096,285 @@
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "TRAIN: [0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [1 1 0 0]\n",
+      "TRAIN: [1 1 0 0 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [0 1 1 1]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [1 1 3 3]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [0 0 3 0]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 3 1 1 0 0 0 0 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [0 1 1 1]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 0 0 0 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [0 3 1 1]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 1 0 0 0 3 0 0 0 1 1 0] TEST: [0 0 0 0]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 3 0 0 0 1 1 0] TEST: [0 1 0 0]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 1 1 0] TEST: [0 3 0 0]\n",
+      "TRAIN: [1 1 0 0 0 1 1 1 1 1 3 3 0 0 3 0 0 1 1 1 0 3 1 1 0 0 0 0 0 1 0 0 0 3 0 0] TEST: [0 1 1 0]\n"
+     ]
+    }
+   ],
    "source": [
-    "cv = StratifiedShuffleSplit(n_splits=30, random_state=0, test_size=10)"
+    "from sklearn.model_selection import KFold\n",
+    "kf = KFold(n_splits=10)\n",
+    "for train_index, test_index in kf.split(connectome_vect):\n",
+    "    print(\"TRAIN:\", groups[train_index], \"TEST:\", groups[test_index])\n",
+    "    groups_train, groups_test = groups[train_index], groups[test_index]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "747eaa15",
+   "metadata": {},
+   "source": [
+    "In the next step, 'cross_val_predict' is used to generate predictions each iteration. 'LinearSCV' will be set as estimator."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 53,
-   "id": "3088ea36",
+   "execution_count": 97,
+   "id": "16361d16",
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/nwiltberg/miniconda3/envs/neuro_ai/lib/python3.7/site-packages/sklearn/model_selection/_split.py:680: UserWarning: The least populated class in y has only 5 members, which is less than n_splits=10.\n",
+      "  UserWarning,\n"
+     ]
+    }
+   ],
    "source": [
-    "scores = []\n",
-    "\n",
-    "for train, test in cv.split(connectome_vect, groups):\n",
-    "    classifier = LinearSVC().fit(connectome_vect[train], groups[train])\n",
-    "    predictions = classifier.predict(connectome_vect[test])\n",
-    "    scores.append(accuracy_score(groups[test], predictions))"
+    "from sklearn.svm import LinearSVC\n",
+    "from sklearn.model_selection import cross_val_predict\n",
+    "est = LinearSVC()\n",
+    "pred = cross_val_predict(est, connectome_vect, groups, cv=10, method='predict')"
```

Comment:
if you set `cv=10` and not indicate the `cv` object you created above, a different `cv` will be applied. The `cross_val_predict` docs outline that if you set a number for `cv` it will run a StratifiedShuffleSplit (IIRC) but you can also set a previously defined `cv` here, e.g.

`my_cv = WhateverCVYouWantToUse`
`pred = cross_val_predict(est, connectome_vect, groups, cv=my_cv, method='predict')`